### ISSUE TITLE
[HOLD #10894 and #9402][Image] Settings - The grey avatar displayed for a brief moment when uploading a picture

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10704]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login at staging.new.expensify.com
2. Navigate to settings by clicking on the user avatar over LHN
3. Click on Profile
4. Click on the avatar edit icon
5. Click on upload photo
6. Select any image of your choice
7. Verify it’s displayed in the avatar icon
8. Click on save

## Expected Result:
Uploaded photo should be displayed without grey color

## Actual Result:
The grey avatar displayed for a brief moment when uploading a picture

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.94.4

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/187550163-04528f46-bdc9-4b2a-bc6a-cc26245dbe74.mp4

https://user-images.githubusercontent.com/93399543/187550182-993cee8a-fea9-4464-94ae-4a6e1169500f.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Luke9389** [comment]
Hey @jasperhuangg should I make this external? I see you worked on refactoring Avatar uploads here: https://github.com/Expensify/Web-Expensify/pull/34346

**Comment by Luke9389** [comment]
bump @jasperhuangg

**Comment by jasperhuangg** [comment]
Hey! I can't seem to reproduce this.. could it be because the internet connection was poor and it was taking a while to load the image from s3? @Luke9389 
 

https://user-images.githubusercontent.com/31285285/188626858-b0162336-a3de-4b0b-a4c1-11f1a2974e83.mov

https://user-images.githubusercontent.com/31285285/188627690-859d8c39-e067-494d-b549-3b8a829f42d2.MP4

**Comment by Luke9389** [comment]
I think that vid you posted of your phone is a reproduction of the issue. The avatar turned grey for a brief moment there.

**Comment by Luke9389** [comment]
I'll investigate whether this can be a front end only change. My hunch is that it will be, because the network will likely never be so fast that it's unnoticeable, and we already have the image on the device, so we should be able to display the image continuously.

**Comment by jasperhuangg** [comment]
> I think that vid you posted of your phone is a reproduction of the issue. The avatar turned grey for a brief moment there.

@Luke9389 that's actually expected to happen, that's the Avatar component switching briefly between the local file and the file in the URL.

**Comment by Luke9389** [comment]
I see. So I think that's the "issue" being reported here. I wonder if there is a way to eliminate that grey period. Perhaps we only switch the avatar source upon loading the next view or something. We could have a contributor look into it, but idk, do you think this is worth keeping open?

**Comment by jasperhuangg** [comment]
Not really, because that's how the behavior was when I was working on the PR, and the PR was reviewed and approved in that state. I don't think this is a bug as much as a flaw with our design. Maybe we can create a follow up issue to work on? But I think that's not as much as a priority now.

@Luke9389

**Comment by Luke9389** [comment]
OK, I think before closing this I'll bring it up in Slack to see if anyone has additional thoughts on what we should do with this issue. Thanks for chiming in @jasperhuangg!

**Comment by jasperhuangg** [comment]
> OK, I think before closing this I'll bring it up in Slack to see if anyone has additional thoughts on what we should do with this issue. Thanks for chiming in @jasperhuangg!

For sure! I think @Beamanator would be a good person to ask since he wrote the DD. Thanks for including me :)

**Comment by Luke9389** [comment]
Putting a hold on this issue until there's a clear, testable resolution here: https://github.com/Expensify/App/issues/9402

**Comment by Beamanator** [comment]
Yeah I must have overlooked the brief flash when testing, I think the main issue we've seen with this is when the previous avatar shows for a while before the new avatar updates (I think only reproduced on Android) - but maybe that's a different issue.

I also think it makes sense to put this on hold (maybe reduce the priority too) until that other issue gets worked on, maybe the solution there will fix this issue 🤞

**Comment by Luke9389** [comment]
This is on HOLD so it shouldn't be marked Overdue.

**Comment by Luke9389** [comment]
This is on HOLD so it shouldn't be marked Overdue.

**Comment by Luke9389** [comment]
This is on HOLD so it shouldn't be marked Overdue.

**Comment by Beamanator** [comment]
Maybe we could also make this monthly :D

**Comment by trjExpensify** [comment]
Coming from this thread, I'm going to assign this to @thomas-coldwell as a follow-up to the caching PR.

**Comment by thomas-coldwell** [comment]
Hi, yes I'll square this off once the image caching PR is merged 🙌

**Comment by Luke9389** [comment]
Still on hold

**Comment by trjExpensify** [comment]
Still held on web image caching that's being held up by a CORS issue.

**Comment by trjExpensify** [comment]
Still held on web image caching that's being held up by a CORS issue.

**Comment by trjExpensify** [comment]
Still on hold for the above to items, which are now held on the server migration.

**Comment by trjExpensify** [comment]
No change, server migration project in progress still.

**Comment by trjExpensify** [comment]
Still on hold for the server migration project.

**Comment by trjExpensify** [comment]
Still on hold for the server migration project.

**Comment by trjExpensify** [comment]
No change.

**Comment by trjExpensify** [comment]
CORS is over the line, so image caching on web is back under construction.

**Comment by trjExpensify** [comment]
Image caching PR was deployed and reverted. Awaiting this now before re-implementing: https://github.com/Expensify/App/issues/32703

**Comment by trjExpensify** [comment]
Damn, we so unlucky Melv! The image caching PR got reverted again.

**Comment by trjExpensify** [comment]
No change.

**Comment by trjExpensify** [comment]
Alrighty.. circling back here. I can no longer reproduce it on upload of a profile picture. So I think we're good to close this out now!
